community/shell/src/main/java/org/neo4j/shell/impl/StandardConsole.java

```diff
@@ -61,6 +62,13 @@ public String readLine( String prompt )
         }
         catch ( IOException e )
         {
+            if (consoleReader != null) {
```

Comment:
you do not need to close your reader that reads from System.in because you will gonna close System.in which you do not want to do